demos/guide/Makefile

```diff
@@ -11,11 +11,13 @@ TESTS = tls-client-block \
         tls-client-non-block \
         quic-server-block \
         quic-server-non-block \
-        quic-client-non-block
+        quic-client-non-block \
+        quic-server-ssl-poll
```

Comment:
This should get added to the build.info file as well, so that the demo builds with enable-demos on the configure line